### DISCUSSION TITLE
KeyError: "None of [Index...are in the [index]"

### DISCUSSION THREAD

**Discussion by mouhamadou-ndim** [Toblerity/Fiona#1110] [Q&A] [answered]
Hello,

I have an error message that prevents me from working with my script.

I want to create a geometry column with this line of code:
`rpg15_19.loc[rpg15_19.geometry_x != None, 'geom'] = rpg15_19['geometry_x']
rpg15_19.loc[rpg15_19.geometry_y != None, 'geom'] = rpg15_19['geometry_y']
rpg15_19.loc[rpg15_19.geometry != None, 'geom'] = rpg15_19['geometry']
`
However, I get an error message like this:

KeyError                                  Traceback (most recent call last)
Input In [5], in <cell line: 2>()
      1 # on créé une seule colonne de géométrie
----> 2 rpg15_19.loc[rpg15_19.geometry_x != None, 'geom'] = rpg15_19['geometry_x']
      3 rpg15_19.loc[rpg15_19.geometry_y != None, 'geom'] = rpg15_19['geometry_y']
      4 rpg15_19.loc[rpg15_19.geometry != None, 'geom'] = rpg15_19['geometry']

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:712, in _LocationIndexer.__setitem__(self, key, value)
    710 else:
    711     key = com.apply_if_callable(key, self.obj)
--> 712 indexer = self._get_setitem_indexer(key)
    713 self._has_valid_setitem_indexer(key)
    715 iloc = self if self.name == "iloc" else self.obj.iloc

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:661, in _LocationIndexer._get_setitem_indexer(self, key)
    659 if isinstance(key, tuple):
    660     with suppress(IndexingError):
--> 661         return self._convert_tuple(key)
    663 if isinstance(key, range):
    664     return list(key)

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:799, in _LocationIndexer._convert_tuple(self, key)
    797     self._validate_key_length(key)
    798     for i, k in enumerate(key):
--> 799         idx = self._convert_to_indexer(k, axis=i)
    800         keyidx.append(idx)
    802 return tuple(keyidx)

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:1291, in _LocIndexer._convert_to_indexer(self, key, axis)
   1289         return inds
   1290     else:
-> 1291         return self._get_listlike_indexer(key, axis)[1]
   1292 else:
   1293     try:

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:1327, in _LocIndexer._get_listlike_indexer(self, key, axis)
   1324 ax = self.obj._get_axis(axis)
   1325 axis_name = self.obj._get_axis_name(axis)
-> 1327 keyarr, indexer = ax._get_indexer_strict(key, axis_name)
   1329 return keyarr, indexer

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexes\base.py:5782, in Index._get_indexer_strict(self, key, axis_name)
   5779 else:
   5780     keyarr, indexer, new_indexer = self._reindex_non_unique(keyarr)
-> 5782 self._raise_if_missing(keyarr, indexer, axis_name)
   5784 keyarr = self.take(indexer)
   5785 if isinstance(key, Index):
   5786     # GH 42790 - Preserve name from an Index

File ~\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexes\base.py:5842, in Index._raise_if_missing(self, key, indexer, axis_name)
   5840     if use_interval_msg:
   5841         key = list(key)
-> 5842     raise KeyError(f"None of [{key}] are in the [{axis_name}]")
   5844 not_found = list(ensure_index(key)[missing_mask.nonzero()[0]].unique())
   5845 raise KeyError(f"{not_found} not in index")

KeyError: "None of [Index([('g', 'e', 'o', 'm', 'e', 't', 'r', 'y', '_', 'x'), ('g', 'e', 'o', 'm', 'e', 't', 'r', 'y', '_', 'x')], dtype='object')] are in the [index]"

can you help me to solve this error ?

**Comment by jorisvandenbossche** [comment, marked as answer]
@Mhdx20 This seems a usage question related to pandas or geopandas, but you posting in the Fiona project (for reading/writing data with GDAL).

I would recommend asking such question at https://stackoverflow.com/. Also, it will help to provide a reproducible example (https://stackoverflow.com/help/minimal-reproducible-example), because right now, without seeing what your data looks like, it is hard to guess what is going wrong.

**Reply by mouhamadou-ndim** [reply]
@jorisvandenbossche 
Thank you very much I will do that
